usecases/rules_usecase.go

```diff
@@ -46,6 +50,26 @@ func (usecase *RuleUsecase) ListRules(ctx context.Context, iterationId string) (
 		})
 }
 
+func (usecase *RuleUsecase) ListRuleExecution(ctx context.Context, testrunId string) ([]models.RuleExecutionStat, error) {
```

Comment:
we'll need to pass in the time boundaries too as filters (presumably coming from the test run limits)